__tests__/Infrastructure/apiRequest.test.ts

```diff
@@ -0,0 +1,899 @@
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+
+import apiRequest from "../../src/infrastructure/apiRequest";
+// Import the mocked function so we can control it in tests
+import isBrowser from "../../src/infrastructure/isBrowser";
+
+import type { RequestConfig } from "../../src/common/common.types";
+
+// Mock the global fetch function
+const mockFetch = vi.fn();
+global.fetch = mockFetch;
```

Comment:
From what I've seen now, this is the only incorrect part since it's mocking fetch instead of using mock service worker

I believe looking at the example on the card may help if AI doesn't, start with the success test

Reply:
The mock service worker library will intercept `fetch` calls, and you will create "fake" endpoints for the test cases